src/sql/workbench/parts/notebook/browser/models/cell.ts

```diff
@@ -329,10 +329,11 @@ export class CellModel implements ICellModel {
 				if ((Array.isArray(content) && content.length > 0) || (!Array.isArray(content) && content)) {
 					// requestExecute expects a string for the code parameter
 					content = Array.isArray(content) ? content.join('') : content;
-					let future = await kernel.requestExecute({
+					const future = kernel.requestExecute({
 						code: content,
 						stop_on_error: true
 					}, false);
+					await future.done;
```

Comment:
Please don't add this. Looking a few lines down you'll see an await on the future.done, but importantly this is after firing a state changed event. Without that, we will potentially mess up the running status / handling of in-progress messages.